### DISCUSSION TITLE
Colors from a continuous Color Axis

### DISCUSSION THREAD

**Discussion by jleviaguirre** [TIBCOSoftware/spotfire-mods#107] [Q&A] [answered]
How do I get these colors from my continuous color axis? The expression only returns one value, so when I run my code, I only get the min color (black)

![image](https://user-images.githubusercontent.com/1098689/208267945-cd4e9d75-8e6c-404a-ada7-78f8b134cc58.png)

      let colors=[];
        (await dataView.allRows()).forEach(row => {
                  colors.push(row.color().hexCode)
        })
        console.log([...new Set(colors)]) //◄ returns ['#000000'], but I expect ['#000000','#FFFFFF']

`

**Comment by objerke** [comment, marked as answer]
My guess is that you only have a single unique value on the color axis. A row can only have a single color from its color scheme. In this case of a gradient scheme, it picks the fist color, which is the min color. 

There is no way to reverse engineer the color scheme from the evaluated data view. What are you trying to accomplish with the colors?

**Reply by jleviaguirre** [reply]
I am developing a [Confusion Matrix Mod](https://github.com/jleviaguirre/js-confusiton-matrix) and want to get the min and max values of the computed cells so I can color them with a gradient.  Instead of using the custom settings for this, I would like to use the Color Axis
